client/jetpack-connect/site-url-input.jsx

```diff
@@ -207,6 +207,10 @@ class JetpackConnectSiteUrlInput extends Component {
 							id="siteSelection"
 							placeholder="Type your site"
 							onChange={ this.handleChange }
+							onSelect={ this.props.onSelect }
+							onSubmit={ this.handleFormSubmit }
```

Comment:
The approach does work, but I don't think it's following the best practices to submit a form from a input field. We should probably wrap the inputs up with `<form></form>` and handle the submission from there. An example could be found at: https://react.dev/reference/react-dom/components/input#reading-the-input-values-when-submitting-a-form